cards/card_component.js

```diff
@@ -80,32 +80,44 @@ BaseCard["{{componentName}}"] = class extends ANSWERS.Component {
     
     cardData.feedbackEnabled = ANSWERS.getAnalyticsOptIn() && cardData.feedback;
 
-    const { details, showMoreDetails } = cardData;
-
-    const cardDetails = details || '';
-    const cardShowMoreConfig = showMoreDetails || {};
-    const { showMoreLimit } = cardShowMoreConfig;
-
-    // Set the value of excessDetailsToggle. Note that it is not enough to have a showMoreLimit.
-    // The card's details must extend past this limit as well for the toggling to be enabled.
-    const showExcessDetailsToggle = showMoreLimit && cardDetails.length > showMoreLimit;
-
-    const truncatedDetails = showExcessDetailsToggle
-      ? `${cardDetails.substring(0, showMoreLimit)}...`
-      : '';
+    const { showExcessDetailsToggle, truncatedDetails } = this._getTruncatedDetails(cardData);
     
     this.validateDataForRender(cardData);
 
     return super.setState({
       ...data,
       card: cardData,
-      showExcessDetailsToggle: showExcessDetailsToggle,
-      truncatedDetails: truncatedDetails,
+      showExcessDetailsToggle,
+      truncatedDetails,
       cardName: `{{componentName}}`,
       relativePath: `{{relativePath}}`
     });
   }
 
+  _getTruncatedDetails({ details = '', showMoreDetails }) {
+    const { showMoreLimit, truncatedDetails: userSpecifiedTruncatedDetails } = showMoreDetails || {};
+    if (userSpecifiedTruncatedDetails) {
+      const showExcessDetailsToggle = userSpecifiedTruncatedDetails.length + 3 < details.length
```

Comment:
it's to prevent the truncated text + ellipsis from being longer than the original text!

Reply:
It makes the assumption that the user wants the suffix to be `...`, which I was a little hesitant about at first since the suffix can be configured in ANSWERS.formatRichText(). However we already force the user to have the suffix be `...` for regular truncated text, so I figured it's probably ok.

Reply:
Would an variable such as `const ellipsesLength` help make the purpose of the number clearer?

Reply:
I thought doing  ANSWERS.formatRichText(blah..) would already include the suffix ellipsis in the string so do we need to do + 3? Is it fine that we assume the suffix will always be the ellipsis (length 3), which is commonly use but we do let user provide their custom suffix?